encoding/gocode/decorators.go

```diff
@@ -244,6 +266,8 @@ func iterateArray(a *dst.ArrayType, existingRawFields map[string]bool) {
 	case *dst.Ident:
 		if existingRawFields[mx.Name] {
 			mx.Name = setStar(mx) + "interface{}"
+		} else if strings.Contains(mx.Name, "_") {
+			mx.Name = withoutUnderscore(mx.Name)
```

Comment:
The issue with these lines is that the `fixRawData` decorator now fix some underscores (but not all) and the `fixUnderscores` decorator fix some raw data 😅 I think we should keep them separated.

I was thinking about something like:
```
iterateArray(a *dst.ArrayType, fn func(i *dst.Ident)) {
   ...
} 
```